### PR TITLE
kubelet startup fail when gpuManager start failed

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:

fail kubelet startup when gpuManager.Start() failed.

**Release note**:

```release-note
Fail kubelet startup when gpuManager.Start() failed.
```

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1317,12 +1317,12 @@ func (kl *Kubelet) initializeModules() error {
 
 	// Start out of memory watcher.
 	if err := kl.oomWatcher.Start(kl.nodeRef); err != nil {
-		return fmt.Errorf("Failed to start OOM watcher %v", err)
+		return fmt.Errorf("failed to start OOM watcher %v", err)
 	}
 
 	// Initialize GPUs
 	if err := kl.gpuManager.Start(); err != nil {
-		glog.Errorf("Failed to start gpuManager %v", err)
```
Comment (resouer) [inline pkg/kubelet/kubelet.go]: There's no reason to block kubelet if `gpuManager` fail to start.

**Comment by vikaschoudhary16** [comment]
GPU manager, I think, will be removed because device manager will be preferred mechanism.

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/kubelet/kubelet.go]: What if we just need the gpu resource on this node?

**Comment by hzxuzhonghu** [comment]
> GPU manager, I think, will be removed because device manager will be preferred mechanism.

Before it is removed, this is necessary I think.

**Pull request closed without merging**